GoRogue/MapGeneration/Region.cs

```diff
@@ -281,46 +302,34 @@ public void RemoveOverlappingPoints(Region imposing)
         /// This is destructive to the region's Connections, so you should try to refrain from generating those
         /// until after you've performed your rotations.
         /// </remarks>
-        public virtual Region Rotate(double degrees, bool doToSelf, Point origin = default)
+        public virtual Region Rotate(double degrees, bool doToSelf, Point origin)
```

Comment:
Any particular reason for this being `virtual`?  It's not bad per-se I'm just curious as to where overloading it would be useful.

Reply:
Because for a test in my game (where I'm using my original implementation of this stuff) I needed to override it and add so some test logic before or after calling `base.Rotate(...)`. 

Another use case is that someone might have a custom method on an object that inherits from region that further transforms the map in some way or adds new collections of points (like `Tracks` or `VisualCues` or `SecretDoors`) or if a user finds out some way they like to rotate the `Connections`